### ISSUE TITLE
no `scikit-sparse` dependency

### ISSUE THREAD

**Issue by jellis18** [nanograv/enterprise#121]
It may be good not to require `scikit-sparse` as a hard dependency since it is only used in special cases (i.e. certain ECORR settings and spatial correlations). We could potentially just make a wrapper like I have done for `mpi4py` in [PAL2](https://github.com/jellis18/PAL2/blob/master/PAL2/nompi4py.pyl) that will still allow the code to run but will use the standard `scipy` library instead. Something like
```python
class cholesky(object):
    
    def __init__(self, x):
        self.cf = scipy.linalg.cho_factor(x)
        
    def __call__(self, y):
        return scipy.linalg.cho_solve(self.cf, y)
    
    def logdet(self):
        return np.sum(2 * np.log(np.diag(self.cf[0])))
```
That way we could import this if `scikit-sparse` is not found and it would have the same signature. We would of course output a warning if this were used but at least it would make installing and running easier.

What do you think?

**Comment by paulthebaker** [comment]
I think this is the right idea.

**Comment by jellis18** [comment]
Implemented in #141
